### PR TITLE
CI: Pre-emptively delete logs dir after cache restore in test-collect-reports

### PR DESCRIPTION
**Pull request by kubawi**

### CODE REVIEW

**Review by VioletHynes — APPROVED** [review]
I don't have the background as to why this might work (you have much more context), but I understand that there's a problem here and trust you :P

All this to say: LGTM!

File: .github/workflows/test-go.yml [hunk 1]
```diff
@@ -476,6 +476,7 @@ jobs:
           name: test-results
           path: test-results/go-test
       - run: |
+          rm -rf test-results/go-test/logs
```
Comment (ncabatoff) [inline .github/workflows/test-go.yml:479]: I think putting this line here would prevent us from having any logs in our test results.  Why not put it instead in sanitize-timing-files, i.e. right after the cache restore?

Comment (ncabatoff) [reply .github/workflows/test-go.yml:479]: I'm reconsidering my alternate suggestion, it doesn't sound quite right either.  Plus, test-collect-reports runs after "Archive test results", so my concern is probably bogus.

**Review by ncabatoff — APPROVED** [review]

File: .github/workflows/test-go.yml (hunk 1 above)
Comment (ncabatoff) [reply .github/workflows/test-go.yml:479]: Yeah it's fine, we have the logs in the archive.

**Pull request merged**